tests/src/test/java/org/eclipse/hono/tests/registry/DeviceRegistrationOptionalAmqpIT.java

```diff
@@ -96,10 +101,26 @@ public static void shutdown(final TestContext ctx) {
     @Test
     public void testAssertRegistrationSucceedsForGateway(final TestContext ctx) {
 
-        deviceRegistryclient.getOrCreateRegistrationClient(Constants.DEFAULT_TENANT)
-            .compose(client -> client.assertRegistration(DeviceRegistrationAmqpIT.DEVICE_ID_ENABLED_WITH_ENABLED_GATEWAY,
-                    DeviceRegistrationAmqpIT.GATEWAY_ID_ENABLED))
-            .setHandler(ctx.asyncAssertSuccess());
+        // Prepare the identities to insert
+        final String deviceId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
+        final String gatewayId = helper.getRandomDeviceId(Constants.DEFAULT_TENANT);
+
+        final Async done = ctx.async();
+        // Insert into the device Registry
+        final Future fut1 = helper.registry.registerDevice(Constants.DEFAULT_TENANT, deviceId,
```

Comment:
Perhaps there two calls can be better merged with the existing calls (using compose())?